hathor/manager.py

```diff
@@ -657,6 +661,7 @@ def propagate_tx(self, tx: BaseTransaction, fails_silently: bool = True) -> bool
             tx.storage = self.tx_storage
         return self.on_new_tx(tx, fails_silently=fails_silently)
 
+    @cpu.profiler('on_new_tx')
```

Comment:
I'm still unsure how to balance the amount the information we want to send to the profiler. You named one good reason to do it but it would "split" the `on_new_tx` into multiple `on_new_tx`.

Reply:
I just did it to check how it would be. Even though there are some rare moments when a specific call to `on_new_tx` is slow, most of the time there are multiple calls that consumes a small CPU individually. See the screenshots below.

# Ungrouped

![image](https://user-images.githubusercontent.com/449119/115942172-c0e3cb00-a45d-11eb-82e1-8463ea826abc.png)

# Grouped
![image](https://user-images.githubusercontent.com/449119/115942223-fdafc200-a45d-11eb-99d7-a3ddf1138031.png)

# Ungrouped with some slow `on_new_tx` (this case happens just occasionally)

![image](https://user-images.githubusercontent.com/449119/115942186-ce995080-a45d-11eb-8c71-635a31a9f837.png)

Reply:
Looking at those screenshots, I don't think it's worth it. If we need to know the tx-ids we could look at the logs.